src/Angor/Client/Pages/Signatures.razor

```diff
@@ -357,26 +437,117 @@
                 sigJson);
 
             FounderProject.LastRequestForSignaturesTime = SignService.SendSignaturesToInvestor(encryptedContent, nostrPrivateKeyHex, signature.investorPubKey, signature.EventId);
-
             Storage.UpdateFounderProject(FounderProject);
 
             signaturesRequests.Single(_ => _.investorPubKey == signature.investorPubKey && _.TimeApproved is null)
                 .TimeApproved = FounderProject.LastRequestForSignaturesTime;
 
             return new OperationResult { Success = true };
-        });
+        }
+        catch (Exception ex)
+        {
+            return new OperationResult { Success = false, Message = $"An error occurred: {ex.Message}" };
+        }
+    }
 
-        if (operationResult.Success)
+
+
+
+
+
+    private async Task ApproveSignature(SignatureRequest signature)
+    {
+        var words = await passwordComponent.GetWalletAsync();
+        if (words == null)
+        {
+            Logger.LogDebug("words is null.");
+            passwordComponent.ShowPassword(async () =>
+            {
+                await ApproveSignature(signature);
+            });
+            return;
+        }
+
+        var operationResult = await PerformSignatureApproval(signature, words);
+
+        if (!operationResult.Success)
         {
+            notificationComponent.ShowErrorMessage(operationResult.Message);
+        }
+        else
+        {
+            StateHasChanged();
+        }
+    }
+
+
+
+
+
+    private async Task ApproveAllSignatures()
+    {
+        isLoading = true;
+        StateHasChanged();
+
+        try
+        {
+            if (passwordComponent.HasPassword())
+            {
+                await ProcessSignatures();
+            }
+            else
+            {
+                passwordComponent.ShowPassword(async () =>
+                {
+                    await ProcessSignatures();
+                });
+            }
+
```

Comment:
I suppose this should go in the finally block, why would we not want to reset password if we fail.